modules/custom/activity_basics/activity_basics.module

```diff
@@ -6,27 +6,51 @@
  */
 
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityPublishedInterface;
 use Drupal\node\NodeInterface;
 use Drupal\group\Entity\GroupContent;
 
 /**
  * Implements hook_entity_insert().
  */
-function activity_basics_entity_insert(EntityInterface $entity) {
+function activity_basics_entity_insert(EntityInterface $entity): void {
   _activity_basics_entity_action($entity, 'create_entitiy_action');
+
+  // Check if the entity created with "published" status.
+  if ($entity instanceof EntityPublishedInterface) {
+    if ($entity->isPublished()) {
+      _activity_basics_entity_action($entity, 'publish_entity_action');
+    }
+  }
 }
 
 /**
  * Implements hook_entity_update().
  */
-function activity_basics_entity_update(EntityInterface $entity) {
+function activity_basics_entity_update(EntityInterface $entity): void {
   _activity_basics_entity_action($entity, 'update_entity_action');
 }
 
+/**
+ * Implements hook_entity_presave().
+ */
+function activity_basics_entity_presave(EntityInterface $entity): void {
+  // Check if the entity become published.
+  if ($entity instanceof EntityPublishedInterface) {
```

Comment:
Also need to check if an entity isn't new because if the entity is new then the original entity will be absent.